nixos/modules/services/networking/netbird-server.nix

```diff
@@ -0,0 +1,336 @@
+      default = { };
+      description = lib.mdDoc "Extra options passed to the IdpManagerConfig.";
+    };
+
+    ports.management = mkOption {
+      type = types.port;
+      default = 8011;
+      description = lib.mdDoc "Internal port of the management server.";
+    };
+
+    ports.signal = mkOption {
+      type = types.port;
+      default = 8012;
+      description = lib.mdDoc "Internal port of the signal server.";
+    };
+
+    fetchOidcConfiguration = mkEnableOption "the automatic fetching of OIDC information.";
+  };
+
+  config = mkIf cfg.enable {
+    services.netbird-server.settings = builtins.mapAttrs (_: mkDefault) {
+      TURN_DOMAIN = cfg.settings.NETBIRD_DOMAIN;
+      TURN_PORT = 3478;
+      TURN_USER = "";
+      TURN_PASSWORD = null;
+
+      NETBIRD_SIGNAL_PROTOCOL = "http";
+      NETBIRD_SIGNAL_PORT = 443;
+
+      NETBIRD_AUTH_USER_ID_CLAIM = "sub";
+      NETBIRD_AUTH_CLIENT_SECRET = "";
+      NETBIRD_MGMT_API_CERT_FILE = null;
+      NETBIRD_MGMT_API_CERT_KEY_FILE = null;
+
+      NETBIRD_MGMT_IDP = "none";
+      NETBIRD_IDP_MGMT_CLIENT_ID = "";
+      NETBIRD_IDP_MGMT_CLIENT_SECRET = "";
+
+      NETBIRD_AUTH_SUPPORTED_SCOPES = [ "openid" "profile" "email" "offline_access" "api" ];
+      NETBIRD_AUTH0_DOMAIN = "";
+      NETBIRD_USE_AUTH0 = true;
+
+      NETBIRD_AUTH_DEVICE_AUTH_ENDPOINT = "";
+    };
+
+    services.netbird-server.managementConfig = with cfg.settings; {
+      Stuns = mkDefault [{
+        Proto = "udp";
+        URI = "stun:${TURN_DOMAIN}:${builtins.toString TURN_PORT}";
+        Username = "";
+        Password = null;
+      }];
+      TURNConfig = {
+        Turns = [{
+          Proto = "udp";
+          URI = "turn:${TURN_DOMAIN}:${builtins.toString TURN_PORT}";
+          Username = TURN_USER;
+          Password = TURN_PASSWORD;
+        }];
+        CredentialsTTL = "12h";
+        Secret = "secret";
+        TimeBasedCredentials = false;
+      };
+      Signal = {
+        Proto = NETBIRD_SIGNAL_PROTOCOL;
+        URI = "${NETBIRD_DOMAIN}:${builtins.toString NETBIRD_SIGNAL_PORT}";
+        Username = "";
+        Password = null;
+      };
+      Datadir = "${stateDir}/data";
+      HttpConfig = {
+        Address = "127.0.0.1:${builtins.toString cfg.ports.management}";
+        AuthIssuer = NETBIRD_AUTH_AUTHORITY;
+        AuthAudience = NETBIRD_AUTH_AUDIENCE;
+        AuthKeysLocation = NETBIRD_AUTH_JWT_CERTS;
+        AuthUserIDClaim = NETBIRD_AUTH_USER_ID_CLAIM;
+        OIDCConfigEndpoint = NETBIRD_AUTH_OIDC_CONFIGURATION_ENDPOINT;
+      };
+      IdpManagerConfig = {
+        ManagerType = NETBIRD_MGMT_IDP;
+        ClientConfig = {
+          Issuer = NETBIRD_AUTH_AUTHORITY;
+          TokenEndpoint = NETBIRD_AUTH_TOKEN_ENDPOINT;
+          ClientID = NETBIRD_IDP_MGMT_CLIENT_ID;
+          ClientSecret = NETBIRD_IDP_MGMT_CLIENT_SECRET;
+          GrantType = "client_credentials";
+        };
+        ExtraConfig = cfg.idpManagerExtraConfig;
```

Comment:
why not just letting folks setting that directly?

Reply:
Are you talking of the grant_type ?